src/sys/windows/named_pipe.rs

```diff
@@ -874,9 +904,25 @@ fn read_done(status: &OVERLAPPED_ENTRY, events: Option<&mut Vec<Event>>) {
         match me.result(status.overlapped()) {
             Ok(n) => {
                 debug_assert_eq!(status.bytes_transferred() as usize, n);
-                buf.set_len(status.bytes_transferred() as usize);
+                // Extend the len depending on the initial len is necessary
+                // when we call `ReadFile` again after resizing
+                // our internal buffer
+                buf.set_len(buf.len() + status.bytes_transferred() as usize);
                 io.read = State::Ok(buf, 0);
             }
+            Err(e) if e.raw_os_error() == Some(ERROR_MORE_DATA as i32) => {
+                match me.remaining_size() {
+                    Ok(rem) => {
+                        buf.set_len(status.bytes_transferred() as usize);
+                        io.read = State::InsufficientBufferSize(buf, rem);
+                        Inner::schedule_read(&me, &mut io, None);
+                        return;
+                    }
+                    Err(e) => {
+                        io.read = State::Err(e);
```

Comment:
I'm not sure about this. This makes it an unrecoverable error, but the original error is not. I feel like this is making the situation worse in case `PeekNamedPipe` returns an error.

Reply:
How is the original error recoverable?

Reply:
The original `ERROR_MORE_DATA` error is not really an error, it we actually read bytes, so it can be ignored.

Reply:
Ah ok, i thought you were referring to the err flow before the change.

Guess we can truncate it in this case